cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -2528,6 +2528,61 @@ pub fn define<'shared>(
             ),
     );
 
+    // Adding with carry
+
+    // XX /r, MR form. Add two GPR registers and set carry flag.
+    recipes.add_template_recipe(
+        EncodingRecipeBuilder::new("rcout", f_binary, 1)
+            .operands_in(vec![gpr, gpr])
+            .operands_out(vec![0])
+            .clobbers_flags(true)
+            .emit(
+                r#"
+                    {{PUT_OP}}(bits, rex2(in_reg0, in_reg1), sink);
+                    modrm_rr(in_reg0, in_reg1, sink);
+                "#,
+            ),
+    );
+
+    // XX /r, MR form. Add two GPR registers and get carry flag.
+    recipes.add_template_recipe(
+        EncodingRecipeBuilder::new("rcin", f_ternary, 1)
+            .operands_in(vec![
+                OperandConstraint::RegClass(gpr),
+                OperandConstraint::RegClass(gpr),
+                OperandConstraint::FixedReg(reg_rflags),
```

Comment:
The carry flags are bools, which cant be stored in rflags if I understand correctly.

Reply:
`cmov` uses `rflags` as the input condition, so I assumed it could be done rather similarly.

Reply:
That one has the clif name of `selectif` which takes an integer flag, not a bool.

Reply:
Right. Looking at this more in detail, it appears that iadd_cin/cout/carry aren't encoded on any platforms, and the instruction definitions (in shared/instruction.rs) tell that they take an in/out b1 operand. In fact, I think this should be an `iflags` operand in this case, because what we really want is to use the flags register as you've done here. Materializing the flag value in an output register would defeat the purpose of this CLIF instruction, at least on x86.

Can you change the 3 instructions, so they take an iflags input/output instead of a b1 value? It might require more changes, like changing the formats, maybe updating tests too. With this change done, the recipes specified here look good.